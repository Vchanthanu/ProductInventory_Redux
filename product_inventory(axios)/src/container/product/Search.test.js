import React from 'react';
import Search from './Search'
import { mount } from 'enzyme';
import { render } from '@testing-library/react'
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import {configure} from 'enzyme'
configure({ adapter: new Adapter() });

it('Rendering Search component',
    () => {
        render(<Search></Search>)
    })

test('matches Search Snapshot', () => {
    const snap = renderer.create(<Search></Search>).toJSON()
    expect(snap).toMatchSnapshot();
})

it('Input Testing',()=>{
    const wrapper = mount(<input type='text' name='productName'></input>)
    const input = wrapper.find('input');
    expect(input.prop('type')).toEqual('text');
    expect(input.prop('name')).toEqual('productName');
})



