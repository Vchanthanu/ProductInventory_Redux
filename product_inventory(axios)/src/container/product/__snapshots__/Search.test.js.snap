// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches Search Snapshot 1`] = `
<div
  className="row"
>
  <div
    className="col-lg-3 col-md-4 col-12"
  >
    <label
      htmlFor="category"
    >
      <b>
        Category
      </b>
    </label>
      
    <select
      id="category"
      name="category"
      onChange={[Function]}
    >
      <option
        value=""
      >
        All
      </option>
      <option
        value="Machinery"
      >
        Machinery
      </option>
      <option
        value="Seed/sapling"
      >
        Seed/Sapling
      </option>
      <option
        value="Tools"
      >
        Tools
      </option>
    </select>
      
  </div>
  <div
    className="col-lg-9 col-md-8 col-12"
  >
    <input
      className="searchbar"
      name="productname"
      onChange={[Function]}
      placeholder="Search for Product using Name..."
      type="text"
    />
  </div>
</div>
`;
